pkg/chartutil/expand.go

```diff
@@ -81,3 +81,67 @@ func ExpandFile(dest, src string) error {
 	defer h.Close()
 	return Expand(dest, h)
 }
+
+// Expand uncompresses and extracts a chart into the specified directory.
+// Return the directory when extracting successfully.
+func ExpandWithPath(dir string, r io.Reader) (string, error) {
+	gr, err := gzip.NewReader(r)
```

Comment:
@bacongobbler No. Have you see technosophos's comment: "The public parts in chartutil are not supposed to change their signatures until Helm 3 so that users do not have to worry about backward compatibility breaks when they use this library. We don't want existing applications to break. So our workaround has been to add new functions instead of changing existing ones. "